notebooks/MINDEvalResults.md

```diff
@@ -0,0 +1,140 @@
+
+```{code-cell} ipython3
+plt.figure(figsize=(12, 3))
+plt.subplots_adjust(wspace=0.3)
+
+plt.subplot(1, 3, 1)
+sns.barplot(data=mind_val_user_metrics, x='recommender', y='NDCG@5')
+plt.xticks(fontsize=8)
+
+plt.subplot(1, 3, 2)
+sns.barplot(data=mind_val_user_metrics, x='recommender', y='NDCG@10')
+plt.xticks(fontsize=8)
+
+plt.subplot(1, 3, 3)
+sns.barplot(data=mind_val_user_metrics, x='recommender', y='MRR')
+plt.xticks(fontsize=8)
+
+plt.show()
+```
+
+```{code-cell} ipython3
+# Function to calculate CI
+def calculate_ci(data, confidence=0.95):
+    se = sem(data)
+    h = se * t.ppf((1 + confidence) / 2, len(data) - 1)
+    return (np.mean(data) - h, np.mean(data) + h)
+
+metrics = ['NDCG@5', 'NDCG@10', 'MRR']
+
+html_tables = ""
+
+# Generate summary tables
+for metric in metrics:
+    summary_table = mind_val_user_metrics.groupby('recommender')[metric].agg(
+        Mean='mean'
+    ).reset_index()
+
+    summary_table['Mean (95% CI)'] = summary_table.apply(
+        lambda row: (
+            f"{row['Mean']:.4f}<br>"
+            f"({calculate_ci(mind_val_user_metrics[mind_val_user_metrics['recommender'] == row['recommender']][metric])[0]:.4f} "
+            f"- {calculate_ci(mind_val_user_metrics[mind_val_user_metrics['recommender'] == row['recommender']][metric])[1]:.4f})"
+        ),
+        axis=1
+    )
+
+    summary_table = summary_table[['recommender', 'Mean (95% CI)']]
+
+    # Add to html_tables string for side by side display
+    html_tables+= f"<div style='display: inline-block; margin-right: 50px;'><h3>Summary Table for {metric}</h3>{summary_table.to_html(index=False, escape=False)}</div>"
+
+display(HTML(html_tables))
+```
+
+### 2. 2 Ranking Overlap Metrics
+
+```{code-cell} ipython3
+plt.figure(figsize=(8, 3))
+
+plt.subplots_adjust(wspace=0.3)
+
+plt.subplot(1, 2, 1)
+sns.barplot(data=mind_val_user_metrics, x='recommender', y='RBO@5')
+plt.xticks(fontsize=8)
+
+plt.subplot(1, 2, 2)
+sns.barplot(data=mind_val_user_metrics, x='recommender', y='RBO@10')
+plt.xticks(fontsize=8)
+
+plt.show()
+```
+
+```{code-cell} ipython3
+metrics = ['RBO@5', 'RBO@10']
+
+html_tables = ""
+
+for metric in metrics:
+    summary_table = mind_val_user_metrics.groupby('recommender')[metric].agg(
+        Mean='mean'
+    ).reset_index()
+
+    summary_table['Mean (95% CI)'] = summary_table.apply(
+        lambda row: (
+            f"{row['Mean']:.4f}<br>"
+            f"({calculate_ci(mind_val_user_metrics[mind_val_user_metrics['recommender'] == row['recommender']][metric])[0]:.4f} "
+            f"- {calculate_ci(mind_val_user_metrics[mind_val_user_metrics['recommender'] == row['recommender']][metric])[1]:.4f})"
+        ),
+        axis=1
+    )
+
+    summary_table = summary_table[['recommender', 'Mean (95% CI)']]
+
+    html_tables+= f"<div style='display: inline-block; margin-right: 50px;'><h3>Summary Table for {metric}</h3>{summary_table.to_html(index=False, escape=False)}</div>"
+
+display(HTML(html_tables))
+```
```

Comment:
I like where this is going to summarize the results.

Style-wise, it would be nice to factor these kinds of things out into library code in the `poprox_recommender` package, so we don't have to repeat it throughout different notebooks.

We might want to use my `eval_tables` code, at least as a starting point — I send you an invite to the repository where that is currently living, and we can vendor it into `poprox_recommender ` for now if we want to.